### ISSUE TITLE
Velero backup v1 custom resource failed since it's trying fetch v1beta1 CR

### ISSUE THREAD

**Issue by Frank51** [vmware-tanzu/velero#5409]
**What steps did you take and what happened:**
[A clear and concise description of what the bug is, and what commands you ran.)
Backup namespace using velero. But backup failed when backing up several custom resources. 
`error="error executing custom action (groupResource=[customresourcedefinitions.apiextensions.k8s.io](http://customresourcedefinitions.apiextensions.k8s.io/), namespace=, name=[mtxengines.matrixx.com](http://mtxengines.matrixx.com/)): rpc error: code = Unknown desc = error fetching v1beta1 version of [mtxengines.matrixx.com](http://mtxengines.matrixx.com/): the server could not find the requested resource" error.file="/remote-source/src/github.com/vmware-tanzu/velero/pkg/backup/item_backupper.go:331"`
Based on the log, looks like velero is trying to get v1beta1 version of CR. But only v1 version CR exist in the cluster.

**What did you expect to happen:**
Backup custom resource using velero should success. 

**The following information will help us better understand what's going on**:

_If you are using velero v1.7.0+:_  
Please use `velero debug  --backup <backupname> --restore <restorename>` to generate the support bundle, and attach to this issue, more options please refer to `velero debug --help` 
The velero log.
https://gist.github.com/Frank51/6c5908cd0cff90ec711e85dfae50aa41

_If you are using earlier versions:_  
Please provide the output of the following commands (Pasting long output into a [GitHub gist](https://gist.github.com) or other pastebin is fine.)
- `kubectl logs deployment/velero -n velero`
- `velero backup describe <backupname>` or `kubectl get backup/<backupname> -n velero -o yaml`
```
Velero's backup: {
    "metadata": {
       "name": "mtx-prod-e1-2022-09-28-17-22-37-mtx-prod-e1",
       "creationTimestamp": null
    },
    "spec": {
       "metadata": {},
       "includedNamespaces": [
          "mtx-prod-e1"
       ],
       "excludedResources": [
          "persistentvolumeclaims",
          "persistentvolumes"
       ],
       "ttl": "262800h0m0s",
       "hooks": {},
       "storageLocation": "storage-location"
    },
    "status": {}
 }
```
- `velero backup logs <backupname>`
- `velero restore describe <restorename>` or `kubectl get restore/<restorename> -n velero -o yaml`
- `velero restore logs <restorename>`

**Anything else you would like to add:**
[Miscellaneous information that will assist in solving the issue.]
https://github.com/vmware-tanzu/velero/issues/4080
Might similar to this issue.

**Environment:**

- Velero version (use `velero version`):  v1.7.1
- Velero features (use `velero client config get features`): 
- Kubernetes version (use `kubectl version`):  Openshift 4.9 (kubernetes 1.22)
- Kubernetes installer & version: 
- Cloud provider or hardware configuration: 
- OS (e.g. from `/etc/os-release`): 

**Vote on this issue!**

This is an invitation to the Velero community to vote on issues, you can see the project's [top voted issues listed here](https://github.com/vmware-tanzu/velero/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc).  
Use the "reaction smiley face" up to the right of this comment to vote.

- :+1: for "I would like to see this bug fixed as soon as possible"
- :-1: for "There are more important bugs to focus on right now"

**Comment by kaovilai** [comment]
```
 time="2022-09-28T14:24:58Z" level=info msg="Found associated CRD mtxengines.matrixx.com to add to backup" backup=velero-ppdm/mtx-prod-e1-2022-09-28-17-22-37-mtx-prod-e1 logSource="pkg/backup/backup.go:480"
 time="2022-09-28T14:24:58Z" level=info msg="Backing up item" backup=velero-ppdm/mtx-prod-e1-2022-09-28-17-22-37-mtx-prod-e1 logSource="pkg/backup/item_backupper.go:121" name=mtxengines.matrixx.com namespace= resource=customresourcedefinitions. time="2022-09-28T14:24:58Z" level=info msg="Executing custom action" backup=velero-ppdm/mtx-prod-e1-2022-09-28-17-22-37-mtx-prod-e1 logSource="pkg/backup/item_backupper.go:327" name=mtxengines.matrixx.com namespace= resource=customresourcedefinitions. time="2022-09-28T14:24:58Z" level=info msg="Executing RemapCRDVersionAction" backup=velero-ppdm/mtx-prod-e1-2022-09-28-17-22-37-mtx-prod-e1 cmd=/velero logSource="pkg/backup/remap_crd_version_action.go:58" pluginName=velero
 time="2022-09-28T14:24:58Z" level=info msg="CustomResourceDefinition mtxengines.matrixx.com appears to be v1beta1, fetching the v1beta version" CRD=mtxengines.matrixx.com backup=velero-ppdm/mtx-prod-e1-2022-09-28-17-22-37-mtx-prod-e1 cmd=/velero logSource="pkg/backup/remap_crd_version_action.go:91" plugin=RemapCRDVersionAction pluginName=velero
 time="2022-09-28T14:24:58Z" level=info msg="1 errors encountered backup up item" backup=velero-ppdm/mtx-prod-e1-2022-09-28-17-22-37-mtx-prod-e1 logSource="pkg/backup/backup.go:431" name=mtxengines.matrixx.com
 time="2022-09-28T14:24:58Z" level=info msg="Found associated CRD network-attachment-definitions.k8s.cni.cncf.io to add to backup" backup=velero-ppdm/mtx-prod-e1-2022-09-28-17-22-37-mtx-prod-e1 logSource="pkg/backup/backup.go:480"
 time="2022-09-28T14:24:58Z" level=info msg="Backing up item" backup=velero-ppdm/mtx-prod-e1-2022-09-28-17-22-37-mtx-prod-e1 logSource="pkg/backup/item_backupper.go:121" name=network-attachment-definitions.k8s.cni.cncf.io namespace= resource=customresourcedefinitions. time="2022-09-28T14:24:58Z" level=info msg="Executing custom action" backup=velero-ppdm/mtx-prod-e1-2022-09-28-17-22-37-mtx-prod-e1 logSource="pkg/backup/item_backupper.go:327" name=network-attachment-definitions.k8s.cni.cncf.io namespace= resource=customresourcedefinitions. time="2022-09-28T14:24:58Z" level=info msg="Executing RemapCRDVersionAction" backup=velero-ppdm/mtx-prod-e1-2022-09-28-17-22-37-mtx-prod-e1 cmd=/velero logSource="pkg/backup/remap_crd_version_action.go:58" pluginName=velero
 time="2022-09-28T14:24:58Z" level=info msg="CustomResourceDefinition network-attachment-definitions.k8s.cni.cncf.io does not appear to be v1beta1, backing up as v1" CRD=network-attachment-definitions.k8s.cni.cncf.io backup=velero-ppdm/mtx-prod-e1-2022-09-28-17-22-37-mtx-prod-e1 cmd=/velero logSource="pkg/backup/remap_crd_version_action.go:97" plugin=RemapCRDVersionAction pluginName=velero
 time="2022-09-28T14:24:58Z" level=info msg="Found associated CRD kopfpeerings.zalando.org to add to backup" backup=velero-ppdm/mtx-prod-e1-2022-09-28-17-22-37-mtx-prod-e1 logSource="pkg/backup/backup.go:480"
 time="2022-09-28T14:24:58Z" level=info msg="Backing up item" backup=velero-ppdm/mtx-prod-e1-2022-09-28-17-22-37-mtx-prod-e1 logSource="pkg/backup/item_backupper.go:121" name=kopfpeerings.zalando.org namespace= resource=customresourcedefinitions. time="2022-09-28T14:24:58Z" level=info msg="Executing custom action" backup=velero-ppdm/mtx-prod-e1-2022-09-28-17-22-37-mtx-prod-e1 logSource="pkg/backup/item_backupper.go:327" name=kopfpeerings.zalando.org namespace= resource=customresourcedefinitions. time="2022-09-28T14:24:58Z" level=info msg="Executing RemapCRDVersionAction" backup=velero-ppdm/mtx-prod-e1-2022-09-28-17-22-37-mtx-prod-e1 cmd=/velero logSource="pkg/backup/remap_crd_version_action.go:58" pluginName=velero
 time="2022-09-28T14:24:58Z" level=info msg="CustomResourceDefinition kopfpeerings.zalando.org appears to be v1beta1, fetching the v1beta version" CRD=kopfpeerings.zalando.org backup=velero-ppdm/mtx-prod-e1-2022-09-28-17-22-37-mtx-prod-e1 cmd=/velero logSource="pkg/backup/remap_crd_version_action.go:91" plugin=RemapCRDVersionAction pluginName=velero
 time="2022-09-28T14:24:58Z" level=info msg="1 errors encountered backup up item" backup=velero-ppdm/mtx-prod-e1-2022-09-28-17-22-37-mtx-prod-e1 logSource="pkg/backup/backup.go:431" name=kopfpeerings.zalando.org
 time="2022-09-28T14:24:58Z" level=info msg="Found associated CRD mtxtras.matrixx.com to add to backup" backup=velero-ppdm/mtx-prod-e1-2022-09-28-17-22-37-mtx-prod-e1 logSource="pkg/backup/backup.go:480"
 time="2022-09-28T14:24:58Z" level=info msg="Backing up item" backup=velero-ppdm/mtx-prod-e1-2022-09-28-17-22-37-mtx-prod-e1 logSource="pkg/backup/item_backupper.go:121" name=mtxtras.matrixx.com namespace= resource=customresourcedefinitions. time="2022-09-28T14:24:58Z" level=info msg="Executing custom action" backup=velero-ppdm/mtx-prod-e1-2022-09-28-17-22-37-mtx-prod-e1 logSource="pkg/backup/item_backupper.go:327" name=mtxtras.matrixx.com namespace= resource=customresourcedefinitions. time="2022-09-28T14:24:58Z" level=info msg="Executing RemapCRDVersionAction" backup=velero-ppdm/mtx-prod-e1-2022-09-28-17-22-37-mtx-prod-e1 cmd=/velero logSource="pkg/backup/remap_crd_version_action.go:58" pluginName=velero
 time="2022-09-28T14:24:58Z" level=info msg="CustomResourceDefinition mtxtras.matrixx.com appears to be v1beta1, fetching the v1beta version" CRD=mtxtras.matrixx.com backup=velero-ppdm/mtx-prod-e1-2022-09-28-17-22-37-mtx-prod-e1 cmd=/velero logSource="pkg/backup/remap_crd_version_action.go:91" plugin=RemapCRDVersionAction pluginName=velero
 time="2022-09-28T14:24:58Z" level=info msg="1 errors encountered backup up item" backup=velero-ppdm/mtx-prod-e1-2022-09-28-17-22-372022-09-28T14:24:58.608853533Z ERROR:  [9445f166267fb52a;977f4acb8de5d8c6] BackupJob: mtx-prod-e1: BackupNamespaceWithVeleroTask: time="2022-09-28T14:24:58Z" level=error msg="Error backing up item" backup=velero-ppdm/mtx-prod-e1-2022-09-28-17-22-37-mtx-prod-e1 error="error executing custom action (groupResource=customresourcedefinitions. the server could not find the requested resource" error.file="/remote-source/src/github.com/vmware-tanzu/velero/pkg/backup/item_backupper.go:331" error.function="github.com/vmware-tanzu/velero/pkg/backup.(*itemBackupper).executeActions" logSource="pkg/backup/backup.go:435" name=mtxsubdomains.matrixx.com
 time="2022-09-28T14:24:58Z" level=error msg="Error backing up item" backup=velero-ppdm/mtx-prod-e1-2022-09-28-17-22-37-mtx-prod-e1 error="error executing custom action (groupResource=customresourcedefinitions. the server could not find the requested resource" error.file="/remote-source/src/github.com/vmware-tanzu/velero/pkg/backup/item_backupper.go:331" error.function="github.com/vmware-tanzu/velero/pkg/backup.(*itemBackupper).executeActions" logSource="pkg/backup/backup.go:435" name=mtxengines.matrixx.com
 time="2022-09-28T14:24:58Z" level=error msg="Error backing up item" backup=velero-ppdm/mtx-prod-e1-2022-09-28-17-22-37-mtx-prod-e1 error="error executing custom action (groupResource=customresourcedefinitions. the server could not find the requested resource" error.file="/remote-source/src/github.com/vmware-tanzu/velero/pkg/backup/item_backupper.go:331" error.function="github.com/vmware-tanzu/velero/pkg/backup.(*itemBackupper).executeActions" logSource="pkg/backup/backup.go:435" name=kopfpeerings.zalando.org
 time="2022-09-28T14:24:58Z" level=error msg="Error backing up item" backup=velero-ppdm/mtx-prod-e1-2022-09-28-17-22-37-mtx-prod-e1 error="error executing custom action (groupResource=customresourcedefinitions. the server could not find the requested resource" error.file="/remote-source/src/github.com/vmware-tanzu/velero/pkg/backup/item_backupper.go:331" error.function="github.com/vmware-tanzu/velero/pkg/backup.(*itemBackupper).executeActions" logSource="pkg/backup/backup.go:435" name=mtxtras.matrixx.com
 time="2022-09-28T14:24:58Z" level=info msg="Backed up a total of 739 items" backup=velero-ppdm/mtx-prod-e1-2022-09-28-17-22-37-mtx-prod-e1 logSource="pkg/backup/backup.go:423" progress=
```

**Comment by kaovilai** [comment]
`CustomResourceDefinition mtxengines.matrixx.com appears to be v1beta1, fetching the v1beta version`

**Comment by kaovilai** [comment]
prior answers https://github.com/vmware-tanzu/velero/issues/4080#issuecomment-1047658192

**Comment by kaovilai** [comment]
Regarding usage of v1beta1 CRD of kopfpeerings.zalando.org, I would recommend transitioning to it's successor project and follow instruction at https://github.com/nolar/kopf/pull/644

**Comment by reasonerjt** [comment]
@Frank51 This should be fixed in v1.8.1 as it has the fix via
https://github.com/vmware-tanzu/velero/pull/4706

**Comment by Frank51** [comment]
@reasonerjt  After upgrading velero to 1.9.1, this problem was fixed. We can close it.

After comparing the CRD yaml of those 3 types, looks like the CRD "mtxengines.matrixx.com" hit "hasSingleVersion" function. So the version of this v1 CRD match "silently upgraded versions" feature.  Backup this CRD failed when velero trying to get v1beta1 CRD of it even though the version is v1.
https://github.com/vmware-tanzu/velero/issues/4080#issuecomment-913627206

**Comment by Lyndon-Li** [comment]
Closing this issue as it is fixed in 1.9.1
